src/main/java/org/folio/service/file/S3JobRunningVerticle.java

```diff
@@ -0,0 +1,329 @@
+
+        updateJobExecutionStatusSafely(
+          queueItem.getJobExecutionId(),
+          new StatusDto()
+            .withErrorStatus(ErrorStatus.FILE_PROCESSING_ERROR)
+            .withStatus(StatusDto.Status.ERROR),
+          params
+        );
+      })
+      .onSuccess((QueueJob result) ->
+        LOGGER.info(
+          "Completed processing job execution {}!",
+          queueItem.getJobExecutionId()
+        )
+      )
+      .onComplete(v -> {
+        queueItemDao.deleteDataImportQueueItem(queueItem.getId());
+
+        File file = localFile.get();
+        if (file != null) {
+          try {
+            FileUtils.delete(file);
+          } catch (IOException e) {
+            LOGGER.error(
+              "Could not clean up temporary file {}: ",
+              file.toPath(),
+              e
+            );
+          }
+        }
+      });
+  }
+
+  protected Future<Void> updateJobExecutionStatusSafely(
+    String jobExecutionId,
+    StatusDto status,
+    OkapiConnectionParams params
+  ) {
+    return uploadDefinitionService
+      .updateJobExecutionStatus(jobExecutionId, status, params)
+      .map((Boolean successful) -> {
+        if (Boolean.FALSE.equals(successful)) {
+          LOGGER.error(
+            "Unable to change job {} status to {}",
+            jobExecutionId,
+            status
+          );
+          throw new IllegalStateException(
+            "Unable to update job execution status"
+          );
+        }
+        return successful;
+      })
+      .mapEmpty();
+  }
+
+  protected Future<QueueJob> downloadFromS3(QueueJob job) {
+    return minioStorageService
+      .readFile(job.getJobExecution().getSourcePath())
+      .map((InputStream inputStream) -> {
+        try (
+          InputStream autoCloseMe = inputStream;
+          OutputStream outputStream = new FileOutputStream(job.getFile())
+        ) {
+          inputStream.transferTo(outputStream);
+
+          return job;
+        } catch (IOException e) {
+          throw new UncheckedIOException(e);
+        }
+      });
+  }
+
+  protected File createLocalFile(DataImportQueueItem queueItem) {
+    try {
+      File localFile = Files
+        .createTempFile(
+          "di-tmp-",
+          // later stage requires correct file extension
+          Path.of(queueItem.getFilePath()).getFileName().toString(),
+          PosixFilePermissions.asFileAttribute(
+            PosixFilePermissions.fromString("rwx------")
+          )
+        )
+        .toFile();
+
+      LOGGER.info("Created temporary file {}", localFile.toPath());
+
+      return localFile;
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  /**
+   * Authenticate and get connection parameters (Okapi URL/token)
+   */
+  protected OkapiConnectionParams getConnectionParams(
+    DataImportQueueItem queueItem
+  ) {
+    OkapiConnectionParams provisionalParams = new OkapiConnectionParams(
+      Map.of(
+        "x-okapi-url",
```

Comment:
consider using org.folio.okapi.common.XOkapiHeaders#URL and others to replace the magic strings contained in this file.